api/src/main/java/org/apache/brooklyn/api/mgmt/ExecutionContext.java

```diff
@@ -64,4 +65,6 @@
 
     boolean isShutdown();
 
+    <T> Maybe<T> getImmediately(Object callableOrSupplier);
```

Comment:
good idea.  we should maybe move `ImmediateSupplier` to the utils package, then we could reference its javadoc?  we might also change the return type to be `ReferenceWithError<Maybe<T>>` so the "can't immediately tell if there's a value" problem state can be detected without throwing.  have marked `@Beta` for now.